GOVERNANCE.md

```diff
@@ -0,0 +1,477 @@
+  Because of this, a github action is installed to lint the licenses of
+  dependencies. This action runs as a normal lint (see "evergreen master") and
+  blocks pull requests if dependencies get imported that do not meet a set of
+  rules agreed upon by the project.
+
+## Release workflow
+
+The following describes how thin-edge.io is released.
+
+The following chapters do not talk about major releases (increasing "x" in a
+"x.y.z" version number) as there is no workflow for these implemented yet.
+
+
+### Preface
+
+The following chapter describes the release process of the thin-edge.io project
+and the corrosponding crates.
+
+There are many ways software projects can be published. In small projects,
+releasing off of the `main` branch might suffice. Larger projects tend to
+implement a dedicated strategy for their release workflow.
+
+As thin-edge.io wants to become a project of potentially many individuals and
+contributing parties, we implement a dedicated release workflow as well.
+
+The goal of the workflow described in this chapter is to have _as little impact
+as possible_ on the general development of the project.
+Releases should _never_ stop, prevent or even just slow down the normal
+development of the project.
+For example, halting development on the `main` branch and forbidding
+contributors and maintainers to merge changes to it even for as little as one
+hour just so that a new release can be published is not acceptable.
+Also should "patch" releases (sometimes called "bugfix releases") have no impact
+on the project pace or even concern developers - they should just happen in
+parallel without any influence on the normal development activities.
+
+With these requirements, we implement the following workflow.
+
+### Semver conformity and API coverage
+
+We adhere to [semver](https://semver.org).
+
+It has to be noted, though, that we also follow the
+[exception in semver compatibility as defined by cargo](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#specifying-dependencies-from-cratesio),
+that specifically means:
+
+> While SemVer says there is no compatibility before 1.0.0, Cargo considers
+> 0.x.y to be compatible with 0.x.z, where y ≥ z and x > 0.
+
+Parts of the API that are covered by semver are explicitly marked. Our public
+API is extensively tested in integration tests. If changes to the public API of
+thin-egde.io are necessary, these changes are documented in the `CHANGELOG.md`
+file of the project, with reasoning why the change was necessary and how users
+can upgrade to the new interface(s).
+
+<!-- TODO: Expand on how exactly these changes are documented? -->
+
+We also define the following rules for our release process:
+
+* We release new features by increasing the "minor" version
+* When we release new patch releases, these releases never contain new features,
+  but only fixes to bugs.
+
+### Release cycle
+
+Minor releases are published roughly every TBD months.
```

Comment:
There is a `TBD` here which we should define.

I'm just throwing some random number in here to get the discussion started: **3 months**.

Reply:
It should be based on content. a minor release can even happen monthly ( we would like to target higher frequency).  What is the need of getting into specific timeline?